citrix_hypervisor/datadog_checks/citrix_hypervisor/check.py

```diff
@@ -0,0 +1,186 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Dict, List
+
+import yaml
+from six.moves import xmlrpc_client as xmlrpclib
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+from .metrics import build_metric
+
+
+class CitrixHypervisorCheck(AgentCheck):
+    __NAMESPACE__ = 'citrix_hypervisor'
+    SERVICE_CHECK_CONNECT = 'can_connect'
+
+    def __init__(self, name, init_config, instances):
+        # type: (str, Dict, List[Dict]) -> None
+        super(CitrixHypervisorCheck, self).__init__(name, init_config, instances)
+
+        self._last_timestamp = 0
+
+        if self.instance.get('url') is None:
+            raise ConfigurationError('Missing configuration option: url')
+        self._base_url = self.instance['url'].rstrip('/')
+
+        self.tags = self.instance.get('tags', [])
+        self.xenserver = None  # type: Any
+        self._additional_tags = []  # type: List[str]
+
+        self.check_initializations.append(self._check_connection)
+
+    def _check_connection(self):
+        # type: () -> None
+        # Get the latest timestamp to reduce the length of the update endpoint response
+        r = self.http.get(self._base_url + '/host_rrd', params={'json': 'true'})
+        if r.status_code == 200:
+            self._last_timestamp = int(float(r.json()['lastupdate'])) - 60
+        else:
+            self.log.warning("Couldn't initialize the timestamp due to HTTP error %s, set it to 0", r.reason)
+
+    def _get_updated_metrics(self):
+        # type: () -> Dict
+        params = {
+            'start': self._last_timestamp,
+            'host': 'true',
+            'json': 'true',
+        }
+        r = self.http.get(self._base_url + '/rrd_updates', params=params)
```

Comment:
I saw you are explicitly login sessions with the username and password provided, just a note that httpwrapper will attempt basic auth if those are encountered in the config, is that ok?

Reply:
`session` is to connect with XMLRPC in order to use the xentools CLI from the integration (`xe host-list` and other commands, to have information about the resource pool master), so there is no choice with the authentication type.
 
However, the other endpoints `/rrd_updates` and `/host_rrd` are basic http endpoint with authentication to get all the host & vms metrics, and the configuration of the HTTP wrapper is enough.
There are 2 interfaces we connect to